### PR TITLE
Fix 124 identifiers sha

### PR DESCRIPTION
**Pull request by valentinedwv**
Add ability to set an identifier per source:
```
      identifierpath: []
      identifiertype: identifiersha
      # identifiertype: ""   # default normalizedjsonsha flight test if we get a different if
      # identifiertype: identifierstring  # needs sanitizing...

```
**identifiertype**
- (normalizedjsonsha) present default is the sha of the normlized json 
  -  (Generate an identifier based on the whole json, if normalize produces ""
- (identifiersha) Generate identifiers based on jsonpath.

**identifierpath** is a jsonpath that gets prefied to the default jsonpath.

Metadata about identifier is written to s3 user metadata.

includes the context repairs since that was the most likely normalize issue.

### CODE REVIEW

File: internal/common/ldproc_test.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+			expected: "_:c14n0 <http://schema.org/name> \"Some type in a graph\" .\n_:c14n0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <bar> .\n",
+			ignore:   false,
+		},
+	}
+	testNormalizeTriple(tests, t)
+}
+
+func testNormalizeTriple(tests []expectations, t *testing.T) {
+	var vipercontext = []byte(`
+context:
```
Comment (nein09) [inline internal/common/ldproc_test.go:85]: Do you need all this stuff in the context here? It would help me understand what the test is doing better if the context only had what was necessary to run the test.

**Comment by nein09** [comment]
I think I understand what this is doing and why, but I think you have a lot of context fixup stuff in this pull request too - would it be too much trouble for you to pull that into a separate PR? I'm getting mixed up as to what changes go with which work.

File: internal/common/ldproc_test.go (hunk 1 above)
Comment (valentinedwv) [reply internal/common/ldproc_test.go:85]: Fixed a bit. Trimmed context, and first test should have been empty. 
Tests used to determine when NormalizeTriples returns an empty document.
 
(Future) Need deeper testing or proxying/wrapping NormalizeTriples of this since empty or same set of triples being returned from normalize causes  identifier conflict.  
Identifier addresses the empty case and hashes the original json to create a more unique id.

**Comment by nein09** [comment]
This seems to be working for me! Ship it.

Thanks!
